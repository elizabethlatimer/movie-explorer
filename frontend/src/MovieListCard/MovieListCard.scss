.MovieListCard {
    width: 90%;
    margin: 10px auto;
    a {
        &:link, &:visited {
            color:black;
            text-decoration:none;
        }
    }
    .card-horizontal {
        display: flex;
        flex: 1 1 auto;
    }
    img {
        max-width: 185px;
    }
    .card-body {
        box-sizing: border-box;
    }
    .line-clamp {
        display:block;
        line-height: 1.5em;
        position:relative;
        height: 9em;
        overflow: hidden;
        &:after {
            content: "";
            text-align: right;
            position: absolute;
            bottom: 0;
            right: 0;
            width: 70%;
            height: 1.5em;
            background: linear-gradient(to right, rgba(255, 255, 255, 0), rgba(255, 255, 255, 1) 80%);

        }
    }

}
