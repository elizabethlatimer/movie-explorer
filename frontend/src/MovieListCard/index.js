import React from 'react';
import { Card } from 'react-bootstrap';
import { Link } from 'react-router-dom';

import './MovieListCard.scss';

function MovieListCard({ movieDetails }) {

  let poster = movieDetails.thumbnailPoster;

  if (movieDetails.thumbnailPoster.includes('null')) {
    poster = 'https://www.joblo.com/assets/images/joblo/database-specific-img-225x333.jpg';
  }

  return (
    <div className="MovieListCard">
    <Link to={`/movies/${movieDetails.id}`}>
      <Card>
        <div className="card-horizontal">
          <div className="img-square-wrapper">
            <img className="" src={poster} alt={`${movieDetails.title} poster`} />
          </div>
          <Card.Body>
            <Card.Title>{movieDetails.title}</Card.Title>
            <Card.Text>Release date: {movieDetails.releaseDate}</Card.Text>
            <Card.Text className='line-clamp'>{movieDetails.overview}</Card.Text>
          </Card.Body>
        </div>
      </Card>
    </Link>
    </div>
  )
}

export default MovieListCard;