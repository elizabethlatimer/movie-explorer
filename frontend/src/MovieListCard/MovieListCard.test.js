import React from 'react';
import { render } from "@testing-library/react";
import MovieListCard from './index';
import {BrowserRouter} from 'react-router-dom'

const testMovie = {
  id: 7,
  title: "test movie",
  overview: "description here",
  releaseDate: "Jan 1, 2021",
  thumbnailPoster:'https://www.joblo.com/assets/images/joblo/database-specific-img-225x333.jpg'
}

it('renders without crashing', () => {
  render(<BrowserRouter><MovieListCard movieDetails={testMovie}/></BrowserRouter>);
});

it('matches the snapshot', () => {
  const {asFragment} = render(<BrowserRouter><MovieListCard movieDetails={testMovie}/></BrowserRouter>);
  expect(asFragment()).toMatchSnapshot();
});