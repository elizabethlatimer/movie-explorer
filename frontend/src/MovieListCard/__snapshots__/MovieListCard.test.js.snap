// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MovieListCard"
  >
    <a
      href="/movies/7"
    >
      <div
        class="card"
      >
        <div
          class="card-horizontal"
        >
          <div
            class="img-square-wrapper"
          >
            <img
              alt="test movie poster"
              class=""
              src="https://www.joblo.com/assets/images/joblo/database-specific-img-225x333.jpg"
            />
          </div>
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              test movie
            </div>
            <p
              class="card-text"
            >
              description here
            </p>
            <p
              class="card-text"
            >
              Release date: Jan 1, 2021
            </p>
          </div>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
